File: backend/src/main/java/com/associazionedisabili/model/Booking.java
package com.associazionedisabili.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(
 name = "bookings",
 uniqueConstraints = {
  @UniqueConstraint(name = "uk_booking_user_event", columnNames = {"user_id", "event_id"})
 }
)
public class Booking {
 @Id
 @GeneratedValue(strategy = GenerationType.IDENTITY)
 private Long id;

 @ManyToOne(fetch = FetchType.LAZY, optional = false)
 @JoinColumn(name = "user_id", nullable = false)
 private User user;

 @ManyToOne(fetch = FetchType.LAZY, optional = false)
 @JoinColumn(name = "event_id", nullable = false)
 private Event event;

 @Column(nullable = false, updatable = false)
 private LocalDateTime createdAt;

 @PrePersist
 protected void onCreate() {
  createdAt = LocalDateTime.now();
 }
}

File: backend/Models/Booking.cs
namespace SoccerDreamFermana.Backend.Models;

public sealed class Booking
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public User User { get; set; } = null!;
    public long EventId { get; set; }
    public Event Event { get; set; } = null!;
    public DateTime CreatedAt { get; set; }
}
